velox/exec/tests/SharedArbitratorTest.cpp

```diff
@@ -382,6 +383,173 @@ class SharedArbitrationTest : public exec::test::HiveConnectorTestBase {
     return queryCtx;
   }
 
+  std::shared_ptr<Task> makeHashJoinTask(
+      const std::vector<RowVectorPtr>& vectors,
+      const std::shared_ptr<core::QueryCtx>& queryCtx,
+      uint32_t numDrivers) {
+    auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+    const auto spillDirectory = exec::test::TempDirectoryPath::create();
+    return AssertQueryBuilder(duckDbQueryRunner_)
+        .config(core::QueryConfig::kSpillEnabled, "true")
+        .config(core::QueryConfig::kJoinSpillEnabled, "true")
+        .queryCtx(queryCtx)
+        .maxDrivers(numDrivers)
+        .plan(PlanBuilder(planNodeIdGenerator)
+                  .values(vectors)
+                  .project({"c0", "c1", "c2"})
+                  .hashJoin(
+                      {"c0"},
+                      {"u0"},
+                      PlanBuilder(planNodeIdGenerator)
+                          .values(vectors)
+                          .project({"c0 AS u0", "c1 AS u1", "c2 AS u2"})
+                          .planNode(),
+                      "",
+                      {"c0", "c1", "c2"},
+                      core::JoinType::kInner)
+                  .planNode())
+        .assertResults(
+            "SELECT t1.c0 AS c0, t1.c1 AS c1, t1.c2 "
+            "FROM tmp t1 "
+            "JOIN tmp t2 "
+            "ON t1.c0 = t2.c0");
+  }
+
+  std::shared_ptr<Task> makeAggregateTask(
```

Comment:
Could you help to replace the existing test cases to use these facilities in followup? Thanks!